### PR TITLE
Feat/#2095 remove local uikit

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

removes local uikit code and instead uses our published self contained npm package
some TS ignores added to not break build (uikit lacking in a couple of types)
transpiling dependency to keep svgs working (will create ticket in uikit for proper build step)

**Type of Change**

- [ ] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [ ] No new UIKit components or changes
- [ ] Feature is minimally responsive
- [ ] Manual testing of UI feature
- [ ] Add copyrights to new files
- [ ] Connected ticket to PR

### CODE REVIEW

**Comment by ciaranschutte** [comment]
392 files changed... what a terrifying number

File: components/Footer.tsx [hunk 1]
```diff
@@ -18,10 +18,10 @@
  */
 
 import * as React from 'react';
-import Footer from 'uikit/Footer';
-import { css } from 'uikit';
+import Footer from '@icgc-argo/uikit//Footer';
```
Comment (demariadaniel) [inline components/Footer.tsx:21]: Two slashes here? `uikit//Footer`

File: components/GdprMessage.tsx [hunk 2]
```diff
@@ -18,14 +18,14 @@
...
-import Typography from 'uikit/Typography';
-import Button from 'uikit/Button';
+import Icon from '@icgc-argo/uikit/Icon';
+import Typography from '@icgc-argo/uikit/Typography';
+import Button from '@icgc-argo/uikit/Button';
 import { LOCAL_STORAGE_GDPR_ACCEPTANCE_KEY } from 'global/constants';
-import Link from 'uikit/Link';
-import HyperLink from 'uikit/Link';
+import Link from '@icgc-argo/uikit/Link';
+import HyperLink from '@icgc-argo/uikit/Link';
```
Comment (demariadaniel) [inline components/GdprMessage.tsx:28]: Why do we have 2 import statements for the same module?
It seems Visual Studio prefers this syntax but it's unclear to me why

Comment (caravinci) [reply components/GdprMessage.tsx:28]: Actually 🤔 in those two lines, it's importing the same `default` as different names... a duplicate import.

**Comment by demariadaniel** [comment]
> 392 files changed... what a terrifying number

Good news is the app seems to run fine locally!

File: stories/uikit.stories.tsx [hunk 3]
```diff
@@ -20,14 +20,14 @@
 import { configure, addDecorator } from '@storybook/react';
 import React from 'react';
 
-import { ThemeProvider } from '../uikit';
+import { ThemeProvider } from '@icgc-argo/uikit';
 
 const req = require.context('../uikit', true, /.stories\.tsx$/);
```
Comment (demariadaniel) [inline stories/uikit.stories.tsx:25]: Does this path need to be updated? `../uikit`?

File: components/pages/ClientError.tsx [hunk 4]
```diff
@@ -19,9 +19,9 @@
 
 import ErrorLayout from 'components/pages/error';
 import React from 'react';
-import { css } from 'uikit';
-import A from 'uikit/Link';
-import Typography from 'uikit/Typography';
+import { css } from '@icgc-argo/uikit';
+import A from '@icgc-argo/uikit/Link';
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply components/pages/ClientError.tsx:23]: @demariadaniel goiing to proceed with `Link` , A has no meaning, and HyperLink makes me feel far too nostalgic for the 90s

File: components/ApplicationRoot.tsx [hunk 5]
```diff
@@ -26,9 +26,10 @@ import { PersistentContext } from 'global/hooks/usePersistentContext';
 import createInMemoryCache from 'global/utils/createInMemoryCache';
 import * as React from 'react';
 import ReactDOM from 'react-dom';
-import { css, ThemeProvider } from 'uikit';
-import Modal from 'uikit/Modal';
-import ToastStack from 'uikit/notifications/ToastStack';
+import { css } from '@icgc-argo/uikit';
+import ThemeProvider from '@icgc-argo/uikit/ThemeProvider';
+import Modal from '@icgc-argo/uikit//Modal';
```
Comment (samrichca) [inline components/ApplicationRoot.tsx:31]: //
```suggestion
import Modal from '@icgc-argo/uikit/Modal';
```

File: components/Footer.tsx [hunk 6]
```diff
@@ -18,10 +18,10 @@
  */
 
 import * as React from 'react';
-import Footer from 'uikit/Footer';
-import { css } from 'uikit';
+import Footer from '@icgc-argo/uikit/Footer';
+import { css } from '@icgc-argo/uikit/';
```
Comment (samrichca) [inline components/Footer.tsx:22]:
```suggestion
import { css } from '@icgc-argo/uikit';
```

**Review by demariadaniel — APPROVED** [review]

File: components/placeholders/ContentError.tsx [hunk 7]
```diff
@@ -18,21 +18,18 @@
  */
 
 import React from 'react';
-import ContentPlaceholder from 'uikit/ContentPlaceholder';
-import Icon from 'uikit/Icon';
+import ContentPlaceholder from '@icgc-argo/uikit/ContentPlaceholder';
+import Icon from '@icgc-argo/uikit//Icon';
```
Comment (samrichca) [inline components/placeholders/ContentError.tsx:22]:
```suggestion
import Icon from '@icgc-argo/uikit/Icon';
```

File: next.config.js [hunk 8]
```diff
@@ -24,8 +24,10 @@ const urlJoin = require('url-join');
 const withImages = require('next-images');
 const path = require('path');
 const Dotenv = require('dotenv-webpack');
+const withTranspileModules = require('next-transpile-modules')(['@icgc-argo/uikit']);
+const withPlugins = require('next-compose-plugins');
 
-module.exports = withImages({
+module.exports = withPlugins([withTranspileModules, [withImages]], {
```
Comment (samrichca) [inline next.config.js:30]: :+1:

Comment (ciaranschutte) [reply next.config.js:30]: ty for review with path updates Sam

**Comment by ciaranschutte** [comment]
holding off for the moment, there is a build issue with uikit project

**Comment by ciaranschutte** [comment]
closing in favour of https://github.com/icgc-argo/platform-ui/pull/2279

**Pull request closed without merging**